File: solid-principles/java/java-dip-notification/Notification.java
public class Notification implements Communication {
    public void sendInfo(User user, String message){
        this.sendNotification(user.getIp(), message);
    }

    public void sendNotification(int ip, String message){
        System.out.println("The message (" + message + ") has been sent to the adress: " + ip);
    }
}

File: solid-principles/csharp/csharp-dip-notification/Notification.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace csharp_haber
{
    public class Notification : Communication
    {
        public void SendInfo(User user, string message){
            this.SendNotification(user.GetIp(), message);
        }

        public void SendNotification(int ip, string message){
            Console.WriteLine("The message (" + message + ") has been sent to the adress: " + ip);
        }
    }    
}